packages/text-field/index.js

```diff
@@ -0,0 +1,367 @@
+    // also the `get disabled` isn't actually used, except in the vanilla component
+    // validity: {
+    //   badInput: boolean,
+    //   valid: boolean,
+    // },
+
+    return {
+      getNativeInput: () => {
+        const input = {
+          validity: {
+            badInput: this.getBadInput(),
+            valid: this.getIsValid(),
+          },
+        };
+
+        // https://stackoverflow.com/a/44913378
+        Object.defineProperty(input, 'value', {
+          get: () => this.state.value,
+          set: (value) => this.setState({foundationValue: value}),
+        });
+
+        return input;
+      },
+    };
+  }
+
+  get labelAdapter() {
+    return {
+      shakeLabel: (shakeLabel) => {
+        const {floatingLabelElement: floatingLabel} = this;
+        if (shakeLabel && floatingLabel && floatingLabel.current) {
+          floatingLabel.current.shake();
+        }
+      },
+      floatLabel: (shouldFloatLabel) => this.setState({shouldFloatLabel}),
+      hasLabel: () => !!this.props.label,
+      getLabelWidth: () => this.state.labelWidth,
+    };
+  }
+
+  get lineRippleAdapter() {
+    return {
+      activateLineRipple: () => this.setState({activeLineRipple: true}),
+      deactivateLineRipple: () => this.setState({activeLineRipple: false}),
+      setLineRippleTransformOrigin: (lineRippleCenter) => this.setState({rippleCenter: lineRippleCenter}),
+    };
+  }
+
+  get notchedOutlineAdapter() {
+    return {
+      notchOutline: () => this.setState({shouldNotchOutline: true}),
+      closeOutline: () => this.setState({shouldNotchOutline: false}),
+      hasOutline: () => !!this.props.outlined,
+    };
+  }
+
+  get helperTextAdapter() {
+    return {
+      showToScreenReader: () =>
+        this.setState({helperTextShowToScreenReader: !this.state.helperTextShowToScreenReader}),
+      setValidity: (isValid) => this.setState({isValid}),
+    };
+  }
+
+  /**
+  * render methods
+  */
+
+  render() {
+    const {
+      label,
+      fullWidth,
+      helperText,
+      outlined,
+      leadingIcon,
+      trailingIcon,
+      textarea,
+    } = this.props;
+
+    const textField = (
+      <div
+        className={this.classes}
+        onClick={() => this.foundation_ && this.foundation_.handleTextFieldInteraction()}
+        onKeyDown={() => this.foundation_ && this.foundation_.handleTextFieldInteraction()}
+        key='text-field-container'
+        ref={this.textFieldElement}
+      >
+        {leadingIcon ? this.renderIcon(leadingIcon) : null}
+        {this.renderInput()}
+        {label && !fullWidth ? this.renderLabel() : null}
+        {outlined ? this.renderNotchedOutline() : null}
+        {!fullWidth && !textarea && !outlined ? this.renderLineRipple() : null}
+        {trailingIcon ? this.renderIcon(trailingIcon) : null}
+      </div>
+    );
+
+    if (helperText) {
+      return ([
+        textField, this.renderHelperText(),
+      ]);
+    }
+    return textField;
+  }
+
+  inputProps = (props) => {
```

Comment:
Nit: Move instance variables like this closer to the top of the class. Also, I don't think this needs to be an arrow function and can instead be an instance method.